io/slow5/slow5.go

```diff
@@ -437,3 +441,50 @@ func Write(headers []Header, reads <-chan Read, output io.Writer) error {
 	}
 	return nil
 }
+
+/******************************************************************************
+Aug 15, 2023
+
+StreamVByte (svb) compression of raw signal strength is used to decrease the
+overall size of records in blow5 files. In my tests using a SQLite database
+that contained both slow5 and fastq files, switching from raw signal TEXT to
+svb compressed BLOBs brought the total database size from 12GB to 7.1GB, a
+~40% reduction in size.
+
+This is the primary method that can be used to decrease the size of storing
+slow5 records in alternative datastores to blow5 (SQL databases, etc).
+
+svb is an integer compression algorithm, which are specialized in compressing
+integer arrays, which perfectly fits raw signal strength data. When converting
+from slow5 to blow5, files are additionally compressed with zstd or zlib on top
+of raw signal compression with svb. Still, svb compression is where most the
+data size saving comes from.
+
+Stream VByte paper: https://doi.org/10.48550/arXiv.1709.08990
+
+Keoni
+
+******************************************************************************/
+
+// SvbCompressRawSignal takes a read and converts its raw signal field to two
+// arrays: a mask array and a data array. Both are needed for decompression.
+func SvbCompressRawSignal(rawSignal []int16) (mask, data []byte) {
```

Comment:
Can we have these functions be methods of Read/another type? As it stands it's not clear to me how a client is supposed to take advantage of them.

Maybe use struct inheritance to define a new CompressedRead class and have a method that goes from CompressedRead -> Read and vice versa.

Reply:
I'm using these functions for taking the rawSignal in and out of an SQL database, so having it be a method doesn't work as well as a raw function. 

`CompressedRead` I think would just be a blow5 read rather than slow5 read, which we SHOULD implement at some point. (there is a whole binary format specifically for doing those compressed reads well, with some real performance improvements)